app/views/certificateModal.html

```diff
@@ -0,0 +1,96 @@
+<div class="modal-header">
+  <button type="button" class="close" ng-click="cancel()" aria-hidden="true">&times;</button>
+  <h4 class="modal-title" id="addClientLabel">
+    <i class="glyphicon glyphicon-th"></i>&nbsp;&nbsp;
+    <span ng-show="update">Edit</span>
+    <span ng-show="!update">Add</span>
+     a certificate</h4>
+</div>
+<form name="form" novalidate>
+  <div class="modal-body">
+
+    <!-- Top Alerts -->
+    <alert ng-repeat="alert in alerts.top" type="alert.type">{{alert.msg}}</alert>
+    <!-- Top Alerts -->
+
+    <div class="panel-body">
+      <a ng-if="downloadKeyLink" download="{{keyName}}" class="btn btn-primary" ng-href="{{downloadKeyLink}}" ng-click="downloadKeyFile()"><i class="glyphicon glyphicon-export"></i> Download Key</a>
+      <a ng-if="downloadCertLink" download="{{certName}}" class="btn btn-primary" ng-href="{{downloadCertLink}}" ng-click="downloadCertFile()"><i class="glyphicon glyphicon-export"></i> Download Cert</a>
+    </div>
+
+    <div class="form-group has-error-key">
+      <span>*</span> Indicates a required field<br />
+    </div>
+
+    <fieldset>
+      <div class="form-group-2columns">
+        <div class="form-group pull-left"  ng-class="{ 'has-error' : ngError.days }">
+          <label><span>*</span> Validity in days [365]</label>
+          <input class="form-control" type="number" ng-model="cert.days" ng-disabled="update" focus="{{!update}}" />
+        </div>
+        <div class="form-group pull-right">
+          <label style="display: block;">Channel Type</label>
+          <div class="btn-group">
+            <label class="btn btn-primary" ng-model="cert.type" btn-radio="'client'">Client</label>
+            <label class="btn btn-primary" ng-model="cert.type" btn-radio="'server'">Server</label>
+          </div>
+        </div>
+      </div>
+
+      <div class="form-group-2columns">
+        <div class="form-group pull-right" ng-class="{ 'has-error' : ngError.clientID }">
+          <label><span>*</span> Country Name (2 letter code) [ZA]</label>
+          <input class="form-control" type="text" ng-model="cert.country" ng-disabled="update" focus="{{!update}}" >
```

Comment:
Please ensure that this field is restricted to 2 chars only and that the chars are upper cased.